src/draw.c

```diff
@@ -335,7 +342,14 @@ static struct colored_layout *layout_init_shared(cairo_t *c, const struct notifi
         } else {
                 width -= 2 * settings.h_padding;
                 width -= 2 * settings.frame_width;
-                if (cl->icon) width -= cairo_image_surface_get_width(cl->icon) + settings.h_padding;
+                if (cl->icon) {
+                        if (settings.text_icon_padding) {
+                                width -= cairo_image_surface_get_width(cl->icon) + settings.text_icon_padding;
+                        }
+                        else {
+                                width -= cairo_image_surface_get_width(cl->icon) + settings.h_padding;
```

Comment:
You could simplify this by adding a function `get_text_icon_padding()` which returns either h_padding or text_icon_padding. By adding that, you don't need if's here.

Reply:
Yeah, the drawing code is pretty rough (and untested)

Reply:
That modularization could also allow the fine tuning controls I mentioned before

Reply:
Yeah it would be good to do that. It's probably best to first modularize it, then add tests and lastly add some fine tuning controls.

Reply:
I could move a few code chunks around to make it a bit more modularized but should I make it part of this PR ?